### ISSUE TITLE
Select - Having `value` 0 (zero) prevents `valueLabel` function from being called

### ISSUE THREAD

**Issue by pascallapradepodform3d** [grommet/grommet#6829]
On the `Select` component, if I provide `0` (zero) as the value, and also provide a `valueLabel` rendering function, the `valueLabel` function is not called, and my value is not rendered as I would like.

### Expected Behavior

If I provide a `valueLabel` to the `Select` component, it should be called for any `value` that I pass to the component.

### Actual Behavior

When I provide a `valueLabel` to the `Select` component, it is not called if the `value` is `0` (which I expect also happens with other falsy values, though I did not test it).

### URL, screen shot, or Codepen exhibiting the issue

Here is minimal reproduction in Codepen: https://codesandbox.io/s/grommet-v2-template-forked-6dn7wd?file=/index.js

It is also reproducible directly on the v2.grommet.io Select component page.

### Steps to Reproduce

1. In the Codepen, notice that the current option is `0`, and the component displays `0`.
2. Notice in the code I have a `valueLabel` function that should display "Value: [the value]".
3. In the rendered component, select a non-0 value.
4. The component changes to display the desired label defined by `valueLabel`.
5. Select 0 again.
6. The label switches back to something other than what is defined by `valueLabel`.

### Your Environment

- Grommet version: 2.32.1 and the version used by v2.grommet.io and the version used by Codepen
- Browser Name and version: Google Chrome 113.0.5672.126 (arm64)
- Operating System and version (desktop or mobile): macOS Ventura 13.4 (arm64)

### My understanding of why this is happening

I looked into the implementation of the Select component here on Github, and I think I have an idea where the issue is happening. I may or may not be correct about this, but hopefully this can help pinpoint and correct the bug.

At lines 235:237 (https://github.com/grommet/grommet/blob/v2.32.2/src/js/components/Select/Select.js#LL235C7-L237C8) we see that an if statement causes a coercion of the user-provided value before `valueLabel` is called:

```javascript
if (valueLabel instanceof Function) {
  if (value) return valueLabel(value);
}
```

As such, a falsy value, such as `0`, will be treated by JS as `false`, and the `valueLabel` function will not be called.

I don't know if this is the intended behavior, blocking all falsy values (in which case maybe it should be documented under the `valueLabel` property, to explain that falsy JS values will prevent the function to be called).

If it is not, then changing the condition to a stricter null and/or undefined check would likely fix the issue while preserving the intended behavior of the nullish guard.

Otherwise, given that the values are user-provided, they could also be passed to the function as-is, without them being checked by Grommet first (that would be my preferred option as a user).

I did not look into the code enough to see if the value could be nullish without the user providing a nullish value themselves, or if the nullish value can be introduced from within the component's logic.

### Workaround

If others like me are having this issue, the following workaround can be used in the meantime:

```javascript
<Select
  // ...
  value={myValue}
  valueLabel={myValue ? myRenderFn : myRenderFn(myValue)}
/>
```

In other words, since `valueLabel` also accepts a node directly, you can call your render function yourself when you know the value you're providing is falsy.
